Xamarin.Forms.Core/DependencyService.cs

```diff
@@ -70,6 +70,16 @@ static void Initialize()
 				assemblies = assemblies.Union(Registrar.ExtraAssemblies).ToArray();
 			}
 
+			Initialize(assemblies);
+		}
+
+		internal static void Initialize(Assembly[] assemblies)
+		{
+			if (s_initialized)
```

Comment:
The only path to the parameterless `Initialize()` already checks `s_initialized` before calling it.

That said, checking it inside of `Initialize()` would be clearer; I'll update it.